further_link/runner/process_handler.py

```diff
@@ -68,10 +76,47 @@ async def start(self, command, work_dir=None, env={}):
             process_env["PWD"] = self.work_dir
             process_env["USER"] = self.user
 
-        # Ensure that DISPLAY is set, so that user can open GUI windows
-        display = get_first_display()
-        if display is not None:
-            process_env["DISPLAY"] = display
+        if novnc:
+            self.display = f":{self.id}"
+
+            proc = await asyncio.create_subprocess_shell(
+                f"/usr/bin/further-link-vnc start {self.id}",
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.PIPE,
+            )
+
+            logging.info(f"Starting further-link-vnc on display '{self.display}'")
+            # Virtual display should be available before executing 'command'
+            await proc.wait()
+
+            self.display_activity_monitor = DisplayActivityMonitor(
+                display_number=self.id
+            )
+
+            async def on_monitor_activity():
+                if not callable(self.on_display_activity):
+                    return
+
+                proc = await asyncio.create_subprocess_shell(
+                    f"/usr/bin/further-link-vnc url {self.id}",
+                    stdout=asyncio.subprocess.PIPE,
+                    stderr=asyncio.subprocess.PIPE,
+                )
+                task = asyncio.create_task(proc.communicate())
+                await task
+                if task.done():
+                    novnc_url, _ = task.result()
+                    novnc_url = novnc_url.decode().strip()
```

Comment:
another helper could go here, getting the url, and parsing it for the port and path
```
connection_details = await vnc_connection_details()
if connection_details:
    await self.on_display_activity(connection_details) # { port, path }
````